lib/config.go

```diff
@@ -514,7 +518,19 @@ func (c *RuntimeConfig) Validate(systemContext *types.SystemContext, onExecution
 
 	// check we do have at least a runtime
 	if _, ok := c.Runtimes[c.DefaultRuntime]; !ok {
-		return errors.New("no default runtime configured")
+		// Set the default runtime to "runc" if default_runtime is not set
+		if c.DefaultRuntime == "" {
```

Comment:
I'm wondering about the reason behind this. Setting `default_runtime = ""` inside the config seems a clear misconfiguration to me, since the value is set but wrong (empty). On the other hand I'd be totally fine with that if we say something is not configured if the value is `""`.

We don't handle this in a consistent way in my point of view. For example we have no default `conmon` path set, and if it's `""`, we fail.

I'm not sure if we can fix these issues without adding workarounds like these, whereas I like the additional checks and we should go for strong defaults.

Reply:
This is for upgrades actually. default_runtime was not a setting and we want to make sure we roll over configurations smoothly.